plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java

```diff
@@ -27,327 +27,289 @@
+        if (db == null) {
+            throw new InvalidObjectException("Database object passed is null");
+        }
+        try {
+            String dbName = db.getName();
+            Node dbNode = this.getNode(dbName);
+            Structure dbStruct = groundReadWrite.createStructure(dbName);
+            LOG.debug("Node and Structure {}, {}", dbNode.getId(), dbStruct.getId());
+            Map<String, GroundType> structVersionAttribs = new HashMap<>();
+            structVersionAttribs.put(dbName, GroundType.STRING);
+            StructureVersion sv = groundReadWrite.createStructureVersion(dbStruct.getName(), dbStruct.getId(),
+                    structVersionAttribs);
+
+            Tag dbTag = new Tag("1.0.0", dbName, JsonUtil.toJSON(db), GroundType.STRING);
+            String reference = db.getLocationUri();
+            HashMap<String, Tag> tags = new HashMap<>();
+            tags.put(dbName, dbTag);
+
+            Map<String, String> referenceParameterMap = db.getParameters();
+            if (referenceParameterMap == null) {
+                referenceParameterMap = new HashMap<String, String>();
+            }
+            List<String> parent = new ArrayList<String>();
+            List<String> versions = groundReadWrite.getLatestVersions(dbName);
+            if (!versions.isEmpty()) {
+                LOG.debug("leaves {}", versions.get(0));
+                parent.add(versions.get(0));
+            }
+            return groundReadWrite.createNodeVersion(dbNode.getName(), tags, sv.getId(), reference,
+                    referenceParameterMap, dbNode.getId());
+        } catch (GroundException e) {
+            LOG.error("Failure to create a database node: {}", e);
+            throw new MetaException(e.getMessage());
+        }
     }
-  }
 
-  Database getDatabase(String dbName) throws NoSuchObjectException {
-    try {
-      List<Long> versions = groundReadWrite.getNodeFactory().getLeaves(dbName);
-      if (versions.isEmpty()) {
-        throw new GroundException("Database node not found: " + dbName);
-      }
+    // Table related functions
+    NodeVersion createTableComponents(Table table) throws InvalidObjectException, MetaException {
+        try {
+            String dbName = table.getDbName();
+            NodeVersion tableNodeVersion = groundTable.createTableNodeVersion(table);
+            Database prevDb = this.getDatabase(dbName);
 
-      NodeVersion latestVersion = groundReadWrite.getNodeVersionFactory().retrieveFromDatabase(versions.get(0));
-      Map<String, Tag> dbTag = latestVersion.getTags();
+            List<String> versions = groundReadWrite.getLatestVersions(dbName);
 
-      return JsonUtil.fromJSON((String) dbTag.get(dbName).getValue(), Database.class);
-    } catch (GroundException e) {
-      throw new NoSuchObjectException(e.getMessage());
+            NodeVersion dbNodeVersion = this.createDatabase(prevDb);
```

Comment:
there are 2 possibilities
1) create a new table in an existing database
2) move or rename a table to a different database
alter table table_a rename to new_database.table_a
if the nodeversion for the new database does not exist we need to create both